### PR TITLE
Simple input tweaks and new Icons

### PR DESCRIPTION
**Pull request by cerinman**
#### Update SimpleInput component and add `dollar` and `percent` icons

These changes are in preparation to update our forms internally to remove auto formatting and saving on blur but also to help with accessibility. 

- Updates `SimpleInput` component with new props `prefix` and  `suffix` intended to replace the `icon`, `rightIcon`, and `handleResetClick` props
- Adds deprecation warnings for `icon`, `rightIcon`, and `handleResetClick` props
- Makes `icon` and `rightIcon` props more useful. I did this before Derek and I realized that using icons for prefix and suffix was problematic due to localization.  I've left in those updates because I felt it wouldn't hurt to have them even if the props are deprecated.
- Fixes elementRef prop.  It was declared but not being used so I fixed that and updated the docs accordingly.
- Adds tests for `SimpleInput`
- Updates `SimpleInput` docs
- Adds new `dollar` icon
- Adds new `percent` icon
- Adds new `deprecateProp` utils function to help with deprecation warnings
- Updates App constants for new icons so that they show up in the `Icon`  docs
- Updates Change log

screen shot 2018-03-31 at 11 25 28 am
screen shot 2018-03-31 at 11 26 44 am
screen shot 2018-03-31 at 11 28 19 am
screen shot 2018-03-31 at 11 28 27 am

### CODE REVIEW

File: docs/components/Changelog.js [hunk 1]
```diff
@@ -7,6 +7,11 @@ class Changelog extends React.Component {
         <h1>Change Log</h1>
 
         <h2>MX React Components V 5.0</h2>
+        <h3>5.1.30</h3>
```
Comment (jtanner) [inline docs/components/Changelog.js]: What happened to `5.1.29`?

File: src/utils/Deprecation.js [hunk 2]
```diff
@@ -3,5 +3,10 @@ module.exports = {
     if (props[primaryColorPropName]) {
       console.warn(`${primaryColorPropName} is deprecated and will be removed in a future release. Use theme instead.`);
     }
+  },
+  deprecateProp (props, deprecatedPropName, replacementPropName, componentDocsURL) {
```
Comment (jtanner) [inline src/utils/Deprecation.js]: I like this 👍

File: src/components/__tests__/SimpleInput-test.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  });
+
+  it('should provide a ref to the input via the elementRef prop', () => {
+    let inputRef;
+
+    const wrapper = mount(<SimpleInput elementRef={ref => inputRef = ref} />);
+
+    expect(inputRef).toEqual(wrapper.instance().elementRef);
+  });
+});
```
Comment (jtanner) [inline src/components/__tests__/SimpleInput-test.js]: Thanks for adding tests 🙌

File: docs/components/Changelog.js (hunk 1 above)
Comment (cerinman) [reply docs/components/Changelog.js]: I'm guessing that Jason published 5.1.29 with out running yarn release and had to release 5.1.29.  That or some one neglected to update the change log/docs on the last release.

**Pull request merged**